common/es/src/main/java/io/apiman/common/es/util/EsConstants.java

```diff
@@ -78,72 +68,189 @@
+    public static final String ES_FIELD_PLAN_DESCRIPTION = "planDescription";
+    public static final String ES_FIELD_PLAN_ID = "planId";
+    public static final String ES_FIELD_PLAN_NAME = "planName";
+    public static final String ES_FIELD_PLAN_VERSION = "planVersion";
+    public static final String ES_FIELD_PLUGIN_ID = "pluginId";
+    public static final String ES_FIELD_POLICY_IMPL = "policyImpl";
+    public static final String ES_FIELD_POLICY_JSON_CONFIG = "policyJsonConfig";
+    public static final String ES_FIELD_PUBLIC_API = "publicAPI";
     public static final String ES_FIELD_PUBLISHED_ON = "publishedOn";
+    public static final String ES_FIELD_REMOTE_ADDR = "remoteAddr";
+    public static final String ES_FIELD_REQUEST_DURATION = "requestDuration";
+    public static final String ES_FIELD_REQUEST_END = "requestEnd";
+    public static final String ES_FIELD_REQUEST_START = "requestStart";
+    public static final String ES_FIELD_RESOURCE = "resource";
+    public static final String ES_FIELD_RESPONSE_CODE = "responseCode";
+    public static final String ES_FIELD_RESPONSE_MESSAGE = "responseMessage";
     public static final String ES_FIELD_RETIRED_ON = "retiredOn";
-    public static final String ES_FIELD_LOCKED_ON = "lockedOn";
-    public static final String ES_FIELD_EXPORTED_ON = "exportedOn";
-    public static final String ES_FIELD_IMPORTED_ON = "importedOn";
+    public static final String ES_FIELD_ROLE_ID = "roleId";
+    public static final String ES_FIELD_STATUS = "status";
+    public static final String ES_FIELD_SUCCESS = "success";
+    public static final String ES_FIELD_TEMPLATE = "template";
+    public static final String ES_FIELD_TYPE = "type";
+    public static final String ES_FIELD_UPDATED_ON = "updatedOn";
+    public static final String ES_FIELD_URL = "url";
+    public static final String ES_FIELD_USER = "user";
+    public static final String ES_FIELD_USERNAME = "username";
+    public static final String ES_FIELD_USER_ID = "userId";
+    public static final String ES_FIELD_VALUE = "value";
+    public static final String ES_FIELD_VERSION = "version";
+    public static final String ES_FIELD_WHAT = "what";
+    public static final String ES_FIELD_WHO = "who";
 
+    // nested es field prefixes
+    public static final String ES_NESTED_FIELD_PREFIX_API_POLICIES = "apiPolicies.";
+    public static final String ES_NESTED_FIELD_PREFIX_CLIENTS = "clients.";
+    public static final String ES_NESTED_FIELD_PREFIX_CONTRACTS = "contracts.";
+    public static final String ES_NESTED_FIELD_PREFIX_GATEWAYS = "gateways.";
+    public static final String ES_NESTED_FIELD_PREFIX_PLANS = "plans.";
+    public static final String ES_NESTED_FIELD_PREFIX_POLICIES = "policies.";
+    public static final String ES_NESTED_FIELD_PREFIX_TEMPLATES = "templates.";
 
+
+    // nested es fields
+    public static final String ES_NESTED_FIELD_API_POLICIES_POLICY_IMPL = ES_NESTED_FIELD_PREFIX_API_POLICIES + ES_FIELD_POLICY_IMPL;
```

Comment:
We can leave this for another time, but I think this probs needs refactoring as well as I am not certain this is very clear. But it's fine for now!